server/session/session.go

```diff
@@ -254,6 +254,12 @@ func (s *Session) close(tx *world.Tx, c Controllable) {
 	s.closeCurrentContainer(tx)
 	s.chunkLoader.Close(tx)
 
+	if ent, ok := s.ent.Entity(tx); ok {
+		if c, ok := ent.(Controllable); ok {
+			c.Wake()
+		}
+	}
+
```

Comment:
No reason for this entire block here. You can simply call `c.Wake()` without calling `s.ent.Entity(tx)`. `c` already is the player.